corpus/accounts/models.py

```diff
@@ -151,3 +155,22 @@ def save(self, *args, **kwargs):
 
     def __str__(self):
         return f"{self.user.first_name} {self.user.last_name} [{self.sig.name}]"
+
+
+class Core(models.Model):
+    user = models.OneToOneField(User, null=False, on_delete=models.CASCADE)
```

Comment:
Since we're decoupling Core and Faculty, I think this can be a OneToOne to `ExecutiveMember` instead. Only Executive Members should have the ability to take core positions.

Reply:
Hey I tried doing that but now I have this issue when I try to makemigrations 

```
It is impossible to add a non-nullable field 'executivemember' to core without specifying a default. This is because the database needs something to populate existing rows.
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit and manually define a default value in models.py.
```

I tried deleting all pre-existing model instances from the db and it still throws the error. How can I fix this? It has bugged me a lot in the past too lol.

Further, I cannot open the Core model menu in the django-admin site. It throws this error page at me:

![image](https://github.com/IEEE-NITK/corpus/assets/113475659/975db7f0-a86e-474a-976f-915d62428edd)

Plx help

Reply:
Do the following:
- Stop all running containers - `docker kill $(docker ps -a -q)`
- Delete the DB volume - `docker volume rm postgres_data`
- (Optional) Prune your system - `docker system prune`
- Restart all containers - `docker compose up --build`

Note that this will nuke your existing data in the DB. If you've created temp users and other things, maybe try to dump the data using [dumpdata](https://docs.djangoproject.com/en/5.0/ref/django-admin/#dumpdata)